pkg/snmp/traps/formatter.go

```diff
@@ -356,6 +361,20 @@ func formatType(variable gosnmp.SnmpPDU) string {
 func formatValue(variable gosnmp.SnmpPDU) interface{} {
 	switch variable.Value.(type) {
 	case []byte:
+		if variable.Type == gosnmp.IPAddress {
+			return common.IPBytesToString(variable.Value.([]byte))
+		} else if variable.Type == gosnmp.OctetString {
+			b := variable.Value.([]byte)
+			if !utilFunc.IsStringPrintable(b) {
```

Comment:
I think we should avoid testing for `IsStringPrintable` since it might lead to inconsistent results.

For example in those two cases where the []bytes represent a MacAddr, the current code will lead to those results:

1/ `[]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00}` translated into `000000000000`
2/ `[]byte{0x50, 0x50, 0x50, 0x50, 0x50, 0x50}` translated into `PPPPPP`

The both cases, we have a MacAddr as `[]bytes`, but we are translating in different ways the two cases which is not expected.